doc/master_to_main_rename.md

```diff
@@ -0,0 +1,16 @@
+# master to main branch rename instructions
+
+The Azure IoT C SDK initially used a branch named `master` as its primary branch.  This was renamed to `main` in November, 2021.
+
+**If you cloned this repo after November, 2021, you can ignore the rest of this document.**
```

Comment:
Sorry for being annoying since you already said you would but just in case the turkey food comma hits you...

Remember to update these :)

Reply:
Fixed.